server/src/main/java/org/apache/uniffle/server/HealthScriptChecker.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.uniffle.server;
+
+import org.apache.hadoop.util.NodeHealthScriptRunner;
+import org.apache.hadoop.util.Shell;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.uniffle.common.exception.RssException;
+
+public class HealthScriptChecker extends Checker {
+  private static final Logger LOG = LoggerFactory.getLogger(HealthScriptChecker.class);
+  private String healthScriptPath;
+  private static final String ERROR_PATTERN = "ERROR";
+  private long scriptTimeout;
+
+  HealthScriptChecker(ShuffleServerConf conf) {
+    super(conf);
+    this.healthScriptPath = conf.getString(ShuffleServerConf.HEALTH_CHECKER_SCRIPT_PATH);
+    if (!NodeHealthScriptRunner.shouldRun(healthScriptPath)) {
+      LOG.error("Rss health check script:" + healthScriptPath + " is not available "
+          + "or doesn't have execute permission, so abort server.");
+      throw new RssException("Health script not available.");
+    }
+    this.scriptTimeout = conf.getLong(ShuffleServerConf.HEALTH_CHECKER_SCRIPT_EXE_TIMEOUT);
+  }
+
+  @Override
+  public boolean checkIsHealthy() {
+    HealthCheckerExitStatus status = HealthCheckerExitStatus.SUCCESS;
+    // always build executor is to load the latest script, the script file can update in need.
+    Shell.ShellCommandExecutor commandExecutor = new Shell.ShellCommandExecutor(new String[] {healthScriptPath}, null, null, scriptTimeout);
+    try {
+      commandExecutor.execute();
```

Comment:
So the ExitCodeException will be thrown when the execution failed?

Reply:
if shell execute stopped exit 1, throw ExitCodeException. I'll add a test case.

Reply:
Got it.